### PR TITLE
add is_zero_divisor_with_annihilator

### PR DESCRIPTION
**Pull request by tthsqe12**

### CODE REVIEW

File: src/MatrixAlgebra.jl [hunk 1]
```diff
@@ -69,6 +69,11 @@ is_zero_divisor(a::MatAlgElem{T}) where T <: RingElement = is_zero_divisor(det(a
 
 is_zero_divisor(a::MatAlgElem{T}) where T <: FieldElement = rank(a) != degree(a)
 
+function is_zero_divisor_with_annihilator(a::MatAlgElem{T}) where T <: RingElement
+   f, b = is_zero_divisor_with_annihilator(det(a))
+   throw(NotImplementedError(:adj, A)) #return f, b*adj(A)
```
Comment (tthsqe12) [inline src/MatrixAlgebra.jl:74]: @thofma how hard would it be to whip up an `adj` implementation?

Comment (thofma) [reply src/MatrixAlgebra.jl:74]: Would be easy, if someone would have taken care of the sign in `pseudo_inverse`:

```

  pseudo_inv(M::MatrixElem{T}) where {T <: RingElement}

  Given a non-singular n\times n matrix M over a ring return a tuple X, d
  consisting of an n\times n matrix X and a denominator d such that MX = dI_n,
  where I_n is the n\times n identity matrix. The denominator will be the
  determinant of M up to sign. If M is singular an exception is raised.
```
Notice the "up to sign".

**Comment by fieker** [comment]
On Thu, Nov 17, 2022 at 07:44:31AM -0800, Tommy Hofmann wrote:
> @thofma commented on this pull request.
> 
> 
> 
> > @@ -69,6 +69,11 @@ is_zero_divisor(a::MatAlgElem{T}) where T <: RingElement = is_zero_divisor(det(a
>  
>  is_zero_divisor(a::MatAlgElem{T}) where T <: FieldElement = rank(a) != degree(a)
>  
> +function is_zero_divisor_with_annihilator(a::MatAlgElem{T}) where T <: RingElement
> +   f, b = is_zero_divisor_with_annihilator(det(a))
> +   throw(NotImplementedError(:adj, A)) #return f, b*adj(A)
> 
> Would be easy, if someone would have taken care of the sign in `pseudo_inverse`:
> 
> ```
> 
>   pseudo_inv(M::MatrixElem{T}) where {T <: RingElement}
> 
>   Given a non-singular n\times n matrix M over a ring return a tuple X, d
>   consisting of an n\times n matrix X and a denominator d such that MX = dI_n,
>   where I_n is the n\times n identity matrix. The denominator will be the
>   determinant of M up to sign. If M is singular an exception is raised.
> ```
> Notice the "up to sign".
I'd like to remove the link to the determinant completely and rather
have the matrix and the denominator coprime

File: src/MatrixAlgebra.jl (hunk 1 above)
Comment (tthsqe12) [reply src/MatrixAlgebra.jl:74]: up to sign isn't a problem, the prohibition against singular matrices is the problem.

Comment (thofma) [reply src/MatrixAlgebra.jl:74]: If you really want the adjugate, then the sign is a problem?

**Comment by thofma** [comment]
There are some non-error branches, that are not tested.

**Pull request merged**